components/script/dom/worker.rs

```diff
@@ -120,3 +120,24 @@ impl<'a> WorkerMethods for JSRef<'a, Worker> {
     event_handler!(message, GetOnmessage, SetOnmessage)
 }
 
+pub struct WorkerMessageHandler {
+    addr: TrustedWorkerAddress,
+    data: *mut u64,
+    nbytes: size_t
+}
+
+impl WorkerMessageHandler {
+    pub fn new(addr: TrustedWorkerAddress, data: *mut u64, nbytes: size_t) -> WorkerMessageHandler {
+        WorkerMessageHandler {
+            addr: addr,
+            data: data,
+            nbytes: nbytes,
+        }
+    }
+}
+
+impl Runnable for WorkerMessageHandler {
+    fn handler(&self){
+        Worker::handle_message(self.addr.clone(), self.data, self.nbytes);
```

Comment:
Also I was not comfortable with using `clone()` here. But the borrow checker always gave me errors.